wgpu/backends/wgpu_native/_ffi.py

```diff
@@ -66,36 +66,41 @@ def get_wgpu_lib_path():
 
     # Load the debug binary if requested
     debug_mode = os.getenv("WGPU_DEBUG", "").strip() == "1"
-    build = "debug" if debug_mode else "release"
-
-    # Get lib filename for supported platforms
-    if sys.platform.startswith("win"):  # no-cover
-        lib_filename = f"wgpu_native-{build}.dll"
-    elif sys.platform.startswith("darwin"):  # no-cover
-        lib_filename = f"libwgpu_native-{build}.dylib"
-    elif sys.platform.startswith("linux"):  # no-cover
-        lib_filename = f"libwgpu_native-{build}.so"
-    else:  # no-cover
-        raise RuntimeError(
-            f"No WGPU library shipped for platform {sys.platform}. Set WGPU_LIB_PATH instead."
-        )
-
-    # Note that this can be a false positive, e.g. ARM linux.
-    embedded_path = get_resource_filename(lib_filename)
-    if not os.path.isfile(embedded_path):  # no-cover
-        env_hint = "You can set the WGPU_LIB_PATH env var to the location of the wgpu-native library."
-        download_hint = _maybe_get_hint_on_download_script().strip()
+    candidate_builds = ["-debug" if debug_mode else "-release"]
+    if not debug_mode:
+        candidate_builds.append("")
+
+    for build in candidate_builds:
+        # Get lib filename for supported platforms
+        if sys.platform.startswith("win"):  # no-cover
+            lib_filename = f"wgpu_native{build}.dll"
+        elif sys.platform.startswith("darwin"):  # no-cover
+            lib_filename = f"libwgpu_native{build}.dylib"
+        elif sys.platform.startswith("linux"):  # no-cover
+            lib_filename = f"libwgpu_native{build}.so"
+        else:  # no-cover
+            raise RuntimeError(
+                f"No WGPU library shipped for platform {sys.platform}. Set WGPU_LIB_PATH instead."
+            )
+
+        try:
+            embedded_path = get_library_filename(lib_filename)
+            return embedded_path
+        except RuntimeError:
+            pass
+    else:
+        main_hint = "Could not find WGPU library libwpgu_native."
         pip_hint = _maybe_get_pip_hint().strip()
-        hints = [pip_hint, download_hint, env_hint]
+        download_hint = _maybe_get_hint_on_download_script().strip()
+        env_hint = "You can set the WGPU_LIB_PATH env var to the location of the wgpu-native library."
+
+        hints = [main_hint, pip_hint, download_hint, env_hint]
         hints = "\n".join([hint for hint in hints if hint])
-        hints = "\n" + hints if hints else ""
-        raise RuntimeError(f"Could not find WGPU library in {embedded_path}.{hints}")
-    else:
-        return embedded_path
+        raise RuntimeError(hints)
 
 
 def _maybe_get_hint_on_download_script():
-    root_dir = os.path.join(get_resource_filename(""), "..", "..")
+    root_dir = get_wgpu_py_root_dir()
     filename = os.path.abspath(
         os.path.join(root_dir, "tools", "download_wgpu_native.py")
     )
```

Comment:
I don't think this path is correct. We can make this bit more explicit, and drop `get_wgpu_py_root_dir`:
```suggestion
    lib_module = sys.modules[__name__.split(".")[0]]
    lib_dir = os.path.abspath(os.path.dirname(lib_module.__file__))
    print(lib_dir)
    filename = os.path.abspath(
        os.path.join(lib_dir, "..", "tools", "download_wgpu_native.py")
    )
```